app/models/concerns/pre_sale_prices/line_item_pre_saleable.rb

```diff
@@ -0,0 +1,20 @@
+module PreSalePrices
+  module LineItemPreSaleable
+    extend ActiveSupport::Concern
+
+    included do
+      before_validation :copy_pre_sale_price
```

Comment:
Yeah I'd send it to Simon for questioning. You'll notice Spree chose to
freeze their price fields too.

On Thu, Jul 7, 2016 at 11:36 AM, Chris Blackburn <notifications@github.com>
wrote:

> In app/models/concerns/pre_sale_prices/line_item_pre_saleable.rb
> <https://github.com/madetech/spree_pre_sale_prices/pull/4#discussion_r69885308>
> :
>
> > @@ -0,0 +1,20 @@
> > +module PreSalePrices
> > +  module LineItemPreSaleable
> > +    extend ActiveSupport::Concern
> > +
> > +    included do
> > +      before_validation :copy_pre_sale_price
>
> Yeah, I could see that argument. What's more correct, the pre-sale price
> when the item was first added to basket, or when the product was purchased?
>
> As you say, the former feels likely more useful for reporting
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/madetech/spree_pre_sale_prices/pull/4/files/2bbe965b78f020763dce578b4fe22098246a6a40#r69885308>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAQpAOilKqeUiW4f-lRqeoIRXbiLXQDkks5qTNafgaJpZM4JG6bg>
> .
>